hasjob/views/helper.py

```diff
@@ -359,6 +359,8 @@ def getposts(basequery=None, pinned=False, showall=False, statuses=None, ageless
         else:
             query = query.order_by(db.desc(JobPost.pinned))
 
+    if return_query:
+        return query
```

Comment:
Since getposts always returns a query, I'd change this to `return_unsorted` or `return_unlimited` because that is what you're skipping.